src/finalfusion/metadata.py

```diff
@@ -0,0 +1,76 @@
+"""
+finalfusion metadata
+"""
+from os import PathLike
+from typing import BinaryIO, Union
+
+import toml
+
+from finalfusion.io import Chunk, ChunkIdentifier, find_chunk, _read_binary, _write_binary
+
+
+class Metadata(dict, Chunk):
+    """
+    Embeddings metadata
+
+    Metadata can be used as a regular Python dict. For serialization, the contents need to be
+    serializable through `toml.dumps`. Finalfusion assumes metadata to be a TOML formatted
+    string.
+
+    Examples
+    --------
+    >>> metadata = Metadata({'Some': 'value', 'number': 1})
+    >>> metadata
+    {'Some': 'value', 'number': 1}
+    >>> metadata['Some']
+    'value'
+    >>> metadata['Some'] = 'other value'
+    >>> metadata['Some']
+    'other value'
+    """
+    @staticmethod
+    def chunk_identifier():
+        return ChunkIdentifier.Metadata
+
+    @staticmethod
+    def read_chunk(file: BinaryIO) -> 'Metadata':
+        file.seek(-12, 1)
+        chunk_id, chunk_len = _read_binary(file, "<IQ")
+        assert ChunkIdentifier(chunk_id) == Metadata.chunk_identifier()
+        return Metadata(toml.loads(file.read(chunk_len).decode("utf-8")))
```

Comment:
The `read` should probably be checked to actually return the given number of bytes. In case the file is written incorrectly, the read size could be less than the indicated size (hitting EOF). Then this would either fail on incorrect TOML (which makes the error hard to understand) or if the cut-off point happens to be valid TOML, it may not fail at all.